### PR TITLE
add parameter to manage dhcp_dir

### PR DESCRIPTION
**Pull request by sethlyons**
#### Pull Request (PR) description
add parameter to manage dhcp_dir

#### This Pull Request (PR) fixes the following issues
n/a

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @sethlyons, can you please rebase against our latest master branch and have a look at the failing spec tests?

**Comment by othalla** [comment]
@sethlyons any news regarding your PR?

File: manifests/pool.pp [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # == Define: dhcp::pool
 #
 define dhcp::pool (
-  Stdlib::IP::Address::V4 $network,
-  Stdlib::IP::Address::V4 $mask,
+  Stdlib::Compat::Ipv4 $network,
```
Comment (ekohl) [inline manifests/pool.pp]: Why are you changing this to the compat type?

File: manifests/pool.pp [hunk 2]
```diff
@@ -15,9 +15,9 @@
...
   $ignore_unknown                           = undef,
 ) {
-  include dhcp::params
 
-  $dhcp_dir = $dhcp::params::dhcp_dir
+  include dhcp
+  $dhcp_dir = $dhcp::dhcp_dir
 
   concat::fragment { "dhcp_pool_${name}":
     target  => "${dhcp_dir}/dhcpd.pools",
```
Comment (ekohl) [inline manifests/pool.pp]:
```suggestion
    target  => "${dhcp::dhcp_dir}/dhcpd.pools",
```

File: manifests/pool.pp [hunk 3]
```diff
@@ -15,9 +15,9 @@
   String $domain_name                       = '',
   $ignore_unknown                           = undef,
 ) {
-  include dhcp::params
 
-  $dhcp_dir = $dhcp::params::dhcp_dir
+  include dhcp
+  $dhcp_dir = $dhcp::dhcp_dir
```
Comment (ekohl) [inline manifests/pool.pp]:
```suggestion
```

File: spec/classes/dhcp_spec.rb [hunk 4]
```diff
@@ -654,4 +654,28 @@
       end
     end
   end
+
+  context 'manage_dhpc_dir=false' do
```
Comment (ekohl) [inline spec/classes/dhcp_spec.rb]:
```suggestion
  context 'manage_dhcp_dir=false' do
```

File: manifests/init.pp [hunk 5]
```diff
@@ -58,7 +58,7 @@
   Hash[String, Hash] $ignoredsubnets                               = {},
   Hash[String, Hash] $pools                                        = {},
   Hash[String, Hash] $pools6                                       = {},
-  Optional[Stdlib::Absolutepath] $dhcpd_binary                     = $dhcp::params::dhcpd_binary
```
Comment (ekohl) [inline manifests/init.pp]: This parameter is still in the service template and shouldn't be removed.

**Comment by dhoppe** [comment]
@sethlyons Are you still interested in this pull request? Then please implement the changes of @ekohl .

**Comment by sethlyons** [comment]
i don't need these changes anymore.  if you think it will help others, i'm happy to clean it up.  otherwise feel free to close it.  please let me know how you'd like me to proceed.

**Comment by ekohl** [comment]
I'm fine with closing. If anyone does need it, they're welcome to submit a patch.

**Comment by bastelfreak** [comment]
Hi. I'm closing this PR due to inactivity. Please reopen and rebase it if you're still interested in it.

**Pull request closed without merging**